### PR TITLE
avoid calling header_hash repeatedly in weight_proof

### PR DESCRIPTION
**Pull request by arvidn**
header_hash is a property that serializes and hash the foliage block, this is somewhat expensive, especially the serialization

### CODE REVIEW

**Review by almogdepaz — DISMISSED** [review]

**Review by almogdepaz — APPROVED** [review]

File: chia/full_node/weight_proof.py [hunk 1]
```diff
@@ -473,6 +475,8 @@ async def __slot_end_vdf(
             tmp_sub_slots_data.append(self.handle_block_vdfs(curr, blocks))
 
             curr = header_blocks[self.blockchain.height_to_hash(uint32(curr.height + 1))]
+            curr_header_hash = curr.header_hash
```
Comment (Nikolaj-K) [inline chia/full_node/weight_proof.py:478]: Naive question: Do you really get a gain if this is set here at the end of the while loop nonetheless?
It also appears that if optimization is indeed a gain here, one could compute the two eos_vdf_iters's upfront.

File: chia/full_node/weight_proof.py [hunk 2]
```diff
@@ -473,6 +475,8 @@ async def __slot_end_vdf(
...
+
         if len(tmp_sub_slots_data) > 0:
```
Comment (Nikolaj-K) [inline chia/full_node/weight_proof.py:480]: this is equivalent to just `if tmp_sub_slots_data:`

File: chia/full_node/weight_proof.py (hunk 1 above)
Comment (almogdepaz) [reply chia/full_node/weight_proof.py:478]: header_hash is a property it calls a function that streams the foliage blocks and then hashes it thats why it helps

Comment (arvidn) [reply chia/full_node/weight_proof.py:478]: I think a better name for it would be `compute_hash()`, and for it to not look like a data member

**Review by mariano54 — APPROVED** [review]

**Pull request merged**